From 1ec3818d999324b686fe7aabb9dbd5e8a81dbd46 Mon Sep 17 00:00:00 2001
From: "Hebbar, Gururaja" <gururaja.hebbar@ti.com>
Date: Tue, 10 Jan 2012 19:14:05 +0530
Subject: [PATCH 0689/1092] arm:omap:am33xx : correct gpio hwmod data

GPIO IP in am335x doesnt support Wakeup. Also, fix incorrect prcm
clock entries for GPIO 2 & 3.

Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 arch/arm/mach-omap2/omap_hwmod_33xx_data.c | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
index a0b6721..6b1ba3c 100644
--- a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
@@ -576,11 +576,10 @@ static struct omap_hwmod_class_sysconfig am33xx_gpio_sysc = {
 	.rev_offs	= 0x0000,
 	.sysc_offs	= 0x0010,
 	.syss_offs	= 0x0114,
-	.sysc_flags	= (SYSC_HAS_AUTOIDLE | SYSC_HAS_ENAWAKEUP |
-			SYSC_HAS_SIDLEMODE | SYSC_HAS_SOFTRESET |
-			SYSS_HAS_RESET_STATUS),
+	.sysc_flags	= (SYSC_HAS_AUTOIDLE | SYSC_HAS_SIDLEMODE |
+			SYSC_HAS_SOFTRESET | SYSS_HAS_RESET_STATUS),
 	.idlemodes	= (SIDLE_FORCE | SIDLE_NO | SIDLE_SMART |
-			SIDLE_SMART_WKUP),
+				SIDLE_SMART_WKUP),
 	.sysc_fields	= &omap_hwmod_sysc_type1,
 };
 
@@ -610,6 +609,7 @@ static struct omap_hwmod_ocp_if *am33xx_gpio0_slaves[] = {
 
 static struct omap_hwmod_opt_clk gpio0_opt_clks[] = {
 	{ .role = "dbclk", .clk = "gpio0_dbclk" },
+	{ .role = "fclk", .clk = "gpio0_fck" },
 };
 
 /* gpio0 */
@@ -646,6 +646,7 @@ static struct omap_hwmod_ocp_if *am33xx_gpio1_slaves[] = {
 
 static struct omap_hwmod_opt_clk gpio1_opt_clks[] = {
 	{ .role = "dbclk", .clk = "gpio1_dbclk" },
+	{ .role = "fclk", .clk = "gpio1_fck" },
 };
 
 static struct omap_hwmod am33xx_gpio1_hwmod = {
@@ -681,6 +682,7 @@ static struct omap_hwmod_ocp_if *am33xx_gpio2_slaves[] = {
 
 static struct omap_hwmod_opt_clk gpio2_opt_clks[] = {
 	{ .role = "dbclk", .clk = "gpio2_dbclk" },
+	{ .role = "fclk", .clk = "gpio2_fck" },
 };
 
 /* gpio2 */
@@ -693,7 +695,7 @@ static struct omap_hwmod am33xx_gpio2_hwmod = {
 	.clkdm_name	= "l4ls_clkdm",
 	.prcm		= {
 		.omap4	= {
-			.clkctrl_offs	= AM33XX_CM_PER_GPIO1_CLKCTRL_OFFSET,
+			.clkctrl_offs	= AM33XX_CM_PER_GPIO2_CLKCTRL_OFFSET,
 			.modulemode	= MODULEMODE_SWCTRL,
 		},
 	},
@@ -717,6 +719,7 @@ static struct omap_hwmod_ocp_if *am33xx_gpio3_slaves[] = {
 
 static struct omap_hwmod_opt_clk gpio3_opt_clks[] = {
 	{ .role = "dbclk", .clk = "gpio3_dbclk" },
+	{ .role = "fclk", .clk = "gpio3_fck" },
 };
 
 /* gpio3 */
@@ -729,7 +732,7 @@ static struct omap_hwmod am33xx_gpio3_hwmod = {
 	.clkdm_name	= "l4ls_clkdm",
 	.prcm		= {
 		.omap4	= {
-			.clkctrl_offs	= AM33XX_CM_PER_GPIO1_CLKCTRL_OFFSET,
+			.clkctrl_offs	= AM33XX_CM_PER_GPIO3_CLKCTRL_OFFSET,
 			.modulemode	= MODULEMODE_SWCTRL,
 		},
 	},
-- 
1.7.11.2

